src/jubjub/mod.rs

```diff
@@ -242,16 +242,26 @@ impl JubjubBls12 {
                 );
             }
 
-            // Check for duplicates, far worse than spec inconsistencies!
+            // Check for simple relations between the generators, that make finding collisions easy;
+            // far worse than spec inconsistencies!
+            let sum = &edwards::Point::zero();
             for (i, p1) in pedersen_hash_generators.iter().enumerate() {
                 if p1 == &edwards::Point::zero() {
                     panic!("Neutral element!");
                 }
-
+                // Used for checking no generator is a sum of previous ones.
+                let sum = &sum.add(&p1,&tmp_params);
```

Comment:
I don't really think this is useful. There are 2<sup>k</sup> - k - 1 possible sums of 2 or more generators chosen from k generators (even ignoring multiples), and this only checks k - 1 sums. The duplicate and perhaps the negation checks are useful because you could imagine those detecting plausible bugs, but I don't think that the sum check does anything to detect plausible bugs.

Reply:
Also, it's not computing a running sum as I think was intended, because `&sum` here always refers to the (immutable) zero `sum` variable outside the loop.

Reply:
Deleted this part of the PR.